packages/jsapi-bootstrap/src/useObjectFetcher.ts

```diff
@@ -0,0 +1,54 @@
+import { VariableDefinition } from '@deephaven/jsapi-types';
+import { useContextOrThrow } from '@deephaven/react-hooks';
+import { createContext } from 'react';
+
+/**
+ * Metadata for an object. Used when needing to fetch an object.
+ */
+export type ObjectMetadata = Record<string, unknown> & VariableDefinition;
+
+export type ObjectFetcher<T = unknown> = (
+  metadata: ObjectMetadata
+) => Promise<T>;
+
+export const ObjectFetcherContext = createContext<ObjectFetcher | null>(null);
+
+/**
+ * Get the metadata for an object definition that is serializable.
+ * @param definition Object definition to get the metadata for
+ * @returns Metadata object that is serializable
+ */
+export function getObjectMetadata(
+  definition: VariableDefinition
+): ObjectMetadata {
+  // Can't use a spread operator because the VariableDefinition JS API object uses property accessors
```

Comment:
Ahh however the JS API implements this it doesn't spread for some reason

Reply:
Yea I'll clean up this comment to be more specific. It's because of the way the JS API creates classes, it uses `Object.defineProperties`... so `JsVariableDefinition` in the outputted code is something like:
```
ciiwcac.JsVariableDefinition_0 = function JsVariableDefinition_0(name_0, type_0){
  this.name_0 = name_0;
  this.type_0 = type_0;
}
;
ciiwcac.JsVariableDefinition_1 = function JsVariableDefinition_1(source){
  jb.$has(source, 'name') && (this.name_0 = jb.$asString(jb.$getAny(source, 'name')));
  jb.$has(source, 'type') && (this.type_0 = jb.$asString(jb.$getAny(source, 'type')));
}
;
defineClass(94, 1, {94:1, 3:1}, ciiwcac.JsVariableDefinition, ciiwcac.JsVariableDefinition_0, ciiwcac.JsVariableDefinition_1);
defineProperties(_, {name:{'get':function getName_3(){
  return ciiwcac.$getName_1(this);
}
```
A simplified example would be:
```
 var a = { foo: 'fiz' }
Object.defineProperty(a, 'bar', { value: 'biz' })
var b = {...a}
console.log(b)

> Object { foo: "fiz" }
```

Tagging @niloc132 but I think he's aware.

Reply:
Actually, there is an `enumerable` attribute you can set when defining a property apparently: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty
I believe the outputted GWT code should set that, seems like it should.

Reply:
I filed a ticket for further discussion on GWT: https://github.com/gwtproject/gwt/issues/9913
It may not be something GWT can change but logged for posterity.